src/main/java/com/github/fppt/jedismock/operations/sortedsets/AbstractZRange.java

```diff
@@ -116,6 +114,18 @@ protected Slice getSliceFromRange(NavigableSet<ZSetEntry> entries) {
         return Response.array(list);
     }
 
+    protected static Stream<Slice> getSliceStream(ZSetEntry e) {
+        int value = e.getScore().intValue();
+        if (abs(value - e.getScore()) < 1e-6) {
```

Comment:
Is `1e-6` toleration is justified by Redis implementation or is it just "a good enough toleration"?

In the first case, we need a comparison test for this (that checks both 1.000001 ->1.000001 and 1.0000001 -> 1.0 cases). But I think this is not the case and what you really need is either 

`if (value % 1 == 0)`

or 

`if (value == Math.rint(value))`

Reply:
Thanks for the example. It seemed to me that the result was rounded to 6 decimal places, but it was just a "effect" of debugger )))